test/hotspot/jtreg/runtime/cds/appcds/jvmti/redefineClasses/OldClassAndRedefineClassApp.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+import java.lang.instrument.ClassDefinition;
+import java.lang.instrument.Instrumentation;
+
+public class OldClassAndRedefineClassApp {
+
+    public static void main(String args[]) throws Throwable {
+
+        System.out.println("Main: loading OldSuper");
+        // Load an old class (version 49), but not linking it.
+        Thread.currentThread().getContextClassLoader().loadClass("OldSuper");
```

Comment:
Why the context classloader? Has it been set in this test?

Reply:
You kind of avoided the question though - why are you using the context classloader instead of e.g the classloader of the current class? Ther CCL is defined as follows:
```
Returns the context ClassLoader for this thread. The context ClassLoader may be set by the creator of the thread for use by code running in this thread when loading classes and resources. If not [set](https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/Thread.html#setContextClassLoader(java.lang.ClassLoader)), the default is to inherit the context class loader from the parent thread.

The context ClassLoader of the primordial thread is typically set to the class loader used to load the application.
```
So if nothing sets it then you are just using the App/System loader, so why not just use that loader directly?

Reply:
I've updated the tests to avoid using `getContextClassLoader()loadClass()`.
For the App/System loader case:
    `ClassLoader.getSystemClassLoader().loadClass()`
For the boot loader case:
    `Class.forName(<class name>, false, null)`